onnx/defs/tensor/defs.cc

```diff
@@ -1085,7 +1085,24 @@ ONNX_OPERATOR_SET_SCHEMA(
             "T",
             OpSchema::all_tensor_types(),
             "Constrain input and output types to all tensor types.")
-        .SetDoc(Upsample_ver7_doc));
+        .SetDoc(Upsample_ver7_doc)
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          if (!hasNInputShapes(ctx, 1)) {
+            return;
+          }
+          propagateShapeAndTypeFromFirstInput(ctx);
+          auto& input_shape = getInputShape(ctx, 0);
+          auto* output_shape = getOutputShape(ctx, 0);
+          auto* scale_attr = ctx.getAttribute("scales");
+          if (input_shape.dim_size() != scale_attr->floats_size())
```

Comment:
Could you add `{}` around the body of if?

Reply:
Sure.